Integrations/integration-CIRCL.yml

```diff
@@ -0,0 +1,371 @@
+description: |-
+  CIRCL Passive DNS is a database storing historical DNS records from various resources.
+  CIRCL Passive SSL is a database storing historical X.509 certificates seen per IP address. The Passive SSL historical data is indexed per IP address.
+configuration:
+- display: Server URL (e.g. https://www.circl.lu)
+  name: url
+  defaultvalue: https://www.circl.lu
+  type: 0
+  required: true
+- display: Username
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Trust any certificate (unsecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |
+    import requests
+    import json
+    import time
+    import datetime
+    import os
+
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    ''' GLOBAL VARS '''
+    BASE_URL = demisto.getParam('url')
+    USERNAME = demisto.getParam('credentials')['identifier']
+    PASSWORD = demisto.getParam('credentials')['password']
+    AUTH = (USERNAME, PASSWORD)
+    USE_SSL = True if demisto.params().get('insecure') else False
+    IS_USING_PROXY = True if demisto.params().get('proxy') else False
+    LAST_TIME_KEY = 'time_last'
+
+    if not IS_USING_PROXY:
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    def http_request(method, url, bad_request_msg = None):
+        responce = requests.request(
+            method,
+            url,
+            auth=AUTH
+        )
+
+        if bad_request_msg is not None and responce.status_code == 400:
+            return_error(bad_request_msg)
+            sys.exit(0)
+        
+        if responce.status_code not in {200, 204}:
+            return_error('Error in API call: [%d] - %s' % (responce.status_code, responce.reason))
+
+        responce.raise_for_status()
+        
+        return responce
+
+    
+    def dns_get_command(url):
+        demisto.results(dns_get(url))
+
+    def dns_get(url):
+        responce = http_dns_get(url)
+
+        results = list(map(lambda line: json.loads(line), responce.text.splitlines()))
+        results = merge_by_rdata(results)
+
+        records = []
+
+        for result in results:
+            records.append(create_dns_record_context(result))
```

Comment:
When I built it, I talked to Asaf and he recommended to leave the time values as they are, because we don't have consistent convention for times, and they might be used for other requests for the same API (although in this API there is currently no filter by time).

I can change the context, or only the human readable. What do you prefer?

Reply:
So we do have a standard for times, and it is not keeping them as timestamps.
I'd prefer to change them to human readable time string please